rpc/server.go

```diff
@@ -25,8 +26,9 @@ func NewServer(conn net.Conn, data types.DataProcessor) *Server {
 	}
 }
 
-func (s *Server) SetMonitorChannel(monitorChan chan struct{}) {
+func (s *Server) SetServerAttributes(monitorChan chan struct{}, addr *string) {
 	s.monitorChan = monitorChan
+	s.clientAddr = addr
```

Comment:
when do you reset it to nil?

Reply:
@vishnuitta in the handlePing() function. Here i'm just assigning the address.